src/scripts/main.js

```diff
@@ -1,3 +1,281 @@
+            - toNumbers(a.children[index].innerHTML)
+        )
+        : rows.sort(
+          (a, b) => toNumbers(a.children[index].innerHTML)
+            - toNumbers(b.children[index].innerHTML)
+        );
+      break;
+  }
+}
+
+tableHead.addEventListener('click', e => {
+  rows = [...tableBody.rows];
+
+  const thIndex = e.target.cellIndex;
+
+  if (e.target.tagName !== 'TH') {
+    return;
+  }
+
+  orders[thIndex] !== 'asc'
+    ? orders[thIndex] = 'asc'
+    : orders[thIndex] = 'dsc';
+
+  for (const index in orders) {
+    if (+index !== thIndex) {
+      orders[index] = 'default';
+    }
+  }
+
+  sortBy(e.target.innerHTML, e.target.cellIndex);
+
+  rows.forEach(row => {
+    tableBody.append(row);
+  });
+});
+
+document.addEventListener('click', e => {
+  const tr = e.target.parentNode;
+  const isActive = tableBody.querySelector('.active');
+
+  if (e.target.tagName !== 'TD') {
+    if (isActive) {
+      isActive.classList.remove('active');
+    }
+
+    return;
+  }
+
+  if (isActive) {
+    isActive.classList.remove('active');
+  }
+
+  tr.classList.add('active');
+});
+
+const createForm = () => {
+  const newForm = document.createElement('form');
+  const newButton = document.createElement('button');
+  const naming = [
+    'Name: ', 'Position: ', 'Office: ', 'Age: ', 'Salary: ',
+  ];
+  const offices = [
+    `Tokyo`, `Singapore`, `London`, `New York`, `Edinburgh`, `San Francisco`,
+  ];
+
+  for (let i = 0; i <= 4; i++) {
+    const type = i <= 2 ? 'name' : 'number';
+    const fieldName = naming[i].replace(': ', '').toLowerCase();
+    const label = document.createElement('label');
+
+    if (i === 2) {
+      const select = document.createElement('select');
+
+      offices.forEach(office => {
+        const option = document.createElement('option');
+
+        option.setAttribute('value', office);
+        option.innerHTML = office;
+        select.append(option);
+      });
+
+      select.setAttribute('name', fieldName);
+      select.required = true;
+      label.innerHTML = naming[i];
+
+      label.append(select);
+    } else {
+      const input = document.createElement('input');
+
+      if (fieldName === 'name') {
+        input.setAttribute('minlength', '4');
+        input.setAttribute('onkeydown', 'return /[a-z ]/i.test(event.key)');
+      };
+
+      if (fieldName === 'age') {
+        input.setAttribute('min', '18');
+        input.setAttribute('max', '90');
+      };
+
+      if (fieldName === 'salary') {
+        input.setAttribute('min', '0');
+      }
+
+      label.innerHTML = naming[i];
+      input.setAttribute('name', fieldName);
+      input.required = true;
+      input.setAttribute('type', type);
+      input.setAttribute('data-qa', fieldName);
+
+      label.append(input);
+    }
+    newForm.append(label);
+  }
+
+  newButton.setAttribute('type', 'submit');
+  newButton.innerHTML = 'Save to table';
+  newForm.append(newButton);
+  newForm.setAttribute('onsubmit', 'return false');
```

Comment:
can you please give me more hints on this one

Reply:
you're using `onClick` event for submitting your form. instead of `submitEvent` read [here](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event)